src/lighting/TrainLighting.cpp

```diff
@@ -0,0 +1,307 @@
+        ADD_PROPERTY(
+                PropertyInfo(
+                        Variant::INT, "light/alternative/source", PROPERTY_HINT_ENUM,
+                        "NotDefined,InternalSource,Transducer,Generator,Accumulator,CurrentCollector,PowerCable,Heater,"
+                        "Main"),
+                "set_alternative_light_source", "get_alternative_light_source");
+        ClassDB::bind_method(
+                D_METHOD("set_alternative_max_voltage", "alternative_max_voltage"),
+                &TrainLighting::set_alternative_max_voltage);
+        ClassDB::bind_method(D_METHOD("get_alternative_max_voltage"), &TrainLighting::get_alternative_max_voltage);
+        ADD_PROPERTY(
+                PropertyInfo(Variant::FLOAT, "light/alternative/max_voltage"), "set_alternative_max_voltage",
+                "get_alternative_max_voltage");
+        ClassDB::bind_method(
+                D_METHOD("set_alternative_light_capacity", "light_source"),
+                &TrainLighting::set_alternative_light_capacity);
+        ClassDB::bind_method(
+                D_METHOD("get_alternative_light_capacity"), &TrainLighting::get_alternative_light_capacity);
+        ADD_PROPERTY(
+                PropertyInfo(Variant::FLOAT, "light/alternative/capacity"), "set_alternative_light_capacity",
+                "get_alternative_light_capacity");
+        ClassDB::bind_method(
+                D_METHOD("set_accumulator_recharge_source", "recharge_source"),
+                &TrainLighting::set_accumulator_recharge_source);
+        ClassDB::bind_method(
+                D_METHOD("get_accumulator_recharge_source"), &TrainLighting::get_accumulator_recharge_source);
+        ADD_PROPERTY(
+                PropertyInfo(
+                        Variant::INT, "source/accumulator/recharge_source", PROPERTY_HINT_ENUM,
+                        "NotDefined,InternalSource,Transducer,Generator,Accumulator,CurrentCollector,PowerCable,Heater,"
+                        "Main"),
+                "set_accumulator_recharge_source", "get_accumulator_recharge_source");
+        ClassDB::bind_method(D_METHOD("set_light_selector_position", "light_selector_position"), &TrainLighting::set_selector_position);
+        ClassDB::bind_method(D_METHOD("get_light_selector_position"), &TrainLighting::get_selector_position);
+        ADD_SIGNAL(MethodInfo(WRAPPER_POSITION_CHANGED_SIGNAL, PropertyInfo(Variant::INT, "position")));
+    }
+
+    void TrainLighting::_do_update_internal_mover(TMoverParameters *mover) {
+        ASSERT_MOVER(mover);
+        TrainPart::_do_update_internal_mover(mover);
+        mover->LightsPosNo = static_cast<int>(light_position_list.size());
+        mover->LightsWrap = wrap_light_selector;
+        mover->LightsDefPos = default_selector_position;
+        mover->LightPower = 0; //LightPower is used there but declared in the Param section in the .fiz file
+        mover->LightPowerSource.SourceType = static_cast<TPowerSource>(light_source);
```

Comment:
should not be casted that way.

Reply:
`No operator= matches arguments of type TPowerSource and TrainController::TrainPowerSource` :)